src/apps/tools/OswAppCalculator.cpp

```diff
@@ -0,0 +1,469 @@
+        i--;
+    }
+    if (i < 0){
+        i = 9;
+    }
+    if (i > 9){
+        i = 0;
+    }
+
+    num1[iNum-1] = i;
+
+    if (hal->btnHasGoneUp(BUTTON_1)){
+        iNum++;
+        i = 0;
+    }
+
+}
+
+void OswAppCalculator::setOP(){
+    OswHal* hal = OswHal::getInstance();
+
+    if (hal->btnHasGoneUp(BUTTON_3)){
+        i++;
+    }
+    if (hal->btnHasGoneUp(BUTTON_2)){
+        i--;
+    }
+    if (i < 0){
+        i = 7;
+    }
+    if (i > 7){
+        i = 0;
+    }
+
+    switch(i){
+        case 0:
+            op = "+";
+            break;
+        case 1:
+            op = "-";
+            break;
+        case 2:
+            op = "*";
+            break;
+        case 3:
+            op = "/";
+            break;
+        case 4:
+            op = "square root";
+            break;
+        case 5:
+            op = "^";
+            break;
+        case 6:
+            op = "x10^";
+            break;
+        case 7:
+            op = "random";
+            break;
+    }
+
+    if (hal->btnHasGoneUp(BUTTON_1)){
+        iNum++;
+        i = 0;
+    }
+
+
+}
+
+void OswAppCalculator::setPNum2(){
+    OswHal* hal = OswHal::getInstance();
+
+    if (hal->btnHasGoneUp(BUTTON_3)){
+        i++;
+    }
+    if (hal->btnHasGoneUp(BUTTON_2)){
+        i--;
+    }
+    if (i < 0){
+        i = 1;
+    }
+    if (i > 1){
+        i = 0;
+    }
+
+    if (i == 0){
+        pNum2 = '+';
+    }else if (i == 1){
+        pNum2 = '-';
+    }
+
+    if (hal->btnHasGoneUp(BUTTON_1)){
+        iNum++;
+        i = 0;
+    }
+
+}
+
+void OswAppCalculator::setNum2(int num2[]){
+    OswHal* hal = OswHal::getInstance();
+
+    if (hal->btnHasGoneUp(BUTTON_3)){
+        i++;
+    }
+    if (hal->btnHasGoneUp(BUTTON_2)){
+        i--;
+    }
+    if (i < 0){
+        i = 9;
+    }
+    if (i > 9){
+        i = 0;
+    }
+
+    num2[iNum-9] = i;
+
+    if (hal->btnHasGoneUp(BUTTON_1)){
+        iNum++;
+        i = 0;
+    }
+
+}
+
+void OswAppCalculator::equal(){
+    OswHal* hal = OswHal::getInstance();
+
+
+    Num1 = num1[0];
+    for (uint8_t i=1;i<6;i++){
+        Num1 = Num1 *10 + num1[i];
+    }
+
+    Num1 = Num1 / 100;
+
+    Num2 = num2[0];
+    for (uint8_t i=1;i<6;i++){
+        Num2 = Num2 *10 + num2[i];
+    }
+
+    Num2 = Num2 / 100;
+
+    if (pNum1 == '-'){
+        Num1 = 0 - Num1;
+    }
+
+    if (pNum2 == '-'){
+        Num2 = 0 - Num2;
+    }
+
+    if (op == "+"){
+        result = Num1 + Num2;
+    }else if (op == "-"){
+        result = Num1 - Num2;
+    }else if (op == "*"){
+        result = Num1 * Num2;
+    }else if (op == "/"){
+        result = Num1 / Num2;
+    }else if (op == "square root"){
+        result = sqrt(Num2);
+    }else if (op == "^"){
+        result = pow(Num1, Num2);
+    }else if (op == "x10^"){
+        result = pow(10, Num2) * Num1;
+    }else if (op == "random" && iNum == 15){
+        result = random(-100000, 100001);
+        result = result / 100;
+        resultS = String(result, 2);
```

Comment:
Why did you write `String`? What's the purpose?

Reply:
result gets converted to a string, because else it would display like that (see image) 
but I removed the unnecessary String(), i didnt know the variable gets converted automatically 
![osw](https://user-images.githubusercontent.com/80181603/192100728-90765847-4662-47fe-88ff-ad40364e0ba6.jpg)